### ISSUE TITLE
6. Payment Gateways: Stripe

### ISSUE THREAD

**Issue by LucHighwalker** [MakeSchool-Tutorials/Proud-Petes-Pet-Emporium#9]
Where the purchase post route gets updated seems to be broken. 

Current (broken id and out of place catch statement)
```
// PURCHASE PET
app.post('/pets/:id/purchase', (req, res) => {
    console.log(req.body);
    // Set your secret key: remember to change this to your live secret key in production
    // See your keys here: https://dashboard.stripe.com/account/apikeys
    var stripe = require("stripe")(process.env.PRIVATE_STRIPE_API_KEY);

    // Token is created using Checkout or Elements!
    // Get the payment token ID submitted by the form:
    const token = req.body.stripeToken; // Using Express

    Pet.findById(req.body.petId).exec((err, pet) => {
      const charge = stripe.charges.create({
        amount: pet.price * 100,
        currency: 'usd',
        description: `Purchased ${pet.name}, ${pet.species}`,
        source: token,
      }).then((chg) => {
        res.redirect(`/pets/${req.params.id}`);
        });
      })
        .catch(err => {
          console.log('Error: ' + err);
        });
    })
  });

```

Should be:
```
// PURCHASE PET
app.post('/pets/:id/purchase', (req, res) => {
    // Set your secret key: remember to change this to your live secret key in production
    // See your keys here: https://dashboard.stripe.com/account/apikeys
    const stripe = require('stripe')(process.env.PRIVATE_STRIPE_API_KEY);

    // Token is created using Checkout or Elements!
    // Get the payment token ID submitted by the form:
    const token = req.body.stripeToken; // Using Express

    Pet.findById(req.params.id).exec((err, pet) => {
      const charge = stripe.charges
        .create({
          amount: pet.price * 100,
          currency: 'usd',
          description: `Purchased ${pet.name}, ${pet.species}`,
          source: token
        })
        .then(chg => {
          res.redirect(`/pets/${req.params.id}`);
        })
        .catch(err => {
          console.log('Error: ' + err);
        });
    });
  });
```

**Comment by ibirnam** [comment]
@LucHighwalker - so could you elaborate on the "broken ID" part? I've looked through my solution (which matches the tutorial) and if you break on that line of code when executing, there's no `id` in `body` to use, only `petID`

<img width="262" alt="screen shot 2019-01-29 at 9 32 44 am" src="https://user-images.githubusercontent.com/1196118/51928428-c3543900-23aa-11e9-9992-725085264046.png">

Secondly, regarding the catch statement, the placement is the standard from [js-beautify](https://beautifier.io/) which VS Code uses automatically, but yours is the more traditional placement, so I've updated the tutorial for that in b1401ee85999657e8c665fe88890284054ab5e0d

**Comment by LucHighwalker** [comment]
@ibirnam Sure, for me the `petId` in the body didn't match any records. It worked at first, but when I seeded the database a few more times after adding a price, the body's `petId` resulted in no matching records. So I changed it to the `params.id`, since it's passed in through the route anyway, and seems to be more reliable.

**Comment by ibirnam** [comment]
@LucHighwalker  - thank you for clarifying, I see what you're saying now!

I've addressed the issue in 8092b1963f57ead7c32428a45f4aaa46e3a23079 for either one to work. Thanks for working through this with me!
